rfc/2020/conformance-validation/mayankshah1607/rfc.md

```diff
@@ -0,0 +1,477 @@
+
+--failFast
+    Do not start new tests after the first test failure.
+
+--parallel int
+    Number of tests to run in parallel
+
+--buildTags String
+    Comma separated list of tags that must be included in the test
+
+--o String
+    Filepath to store the result. Ignored in case of Sonobuoy 
+
+--skipInstall
+    Skips the `linkerd install` process. This is useful for testers who already have linkerd installed prior to running the tests
+
+--addon-config String
+    Filepath to the addon config file. This flag must be used only in "default mode"
+
+--addon-config-sonobuoy String
+    Filepath to the addon config file. This flag must be used only for Sonobuoy. The value of this command is passed to `kubectl cp` command to copy the addon config file to the Sonobuoy pod
+
+--workloadNamespace String
+    This could be a future possibility. The value of this flag identifies the namespace of the workloads against which the conformance tests must run.
+```
+Internally, the script builds a `go test` command by simply appending these values to it. The test configuration specific flags (such as `--wait`,`--enableTests`,`--workloadNamespace`, etc) are passed down to the test suite using the _flags_ go module. These values may be stored in variables (instance of objects, for e.g `TestOptions`) so that tests may run accordingly. A universal `TestHelper` object may also be used to make the flag management process easier.
+
+**(Essential `linkerd install` flags to accept)**
+
+- `--identity-trust-anchors-file` 
+-   `--identity-issuer-certificate-file`
+-  `--identity-issuer-key-file`
+-  `--identity-issuance-lifetime`
+-  `--proxy-cpu-limit`
+-  `--proxy-cpu-request`
+-  `--proxy-memory-limit`
+-  `--proxy-memory-request`
+-  `--ha`
+-  `--addon-config`
+
+The script also builds a `linkerd install` command by simply appending these flags to it. The built command is then executed to install linkerd on the cluster.
+
+Sample Usage :
+
+```bash
+./run.sh --wait=30s \
+    --enableTest="TestAutoProxyInjection/*" \
+    --failFast \
+    --identity-trust-anchors-file=ca.crt \
+    --identity-issuer-certificate-file=issuer.crt \
+    --identity-issuer-key-file=issuer.key \
+    --identity-issuance-lifetime=1m \
+    --proxy-cpu-limit=1 \
+    --proxy-cpu-request=100m \
+    --proxy-memory-limit=250Mi \
+    --proxy-memory-request=20Mi \
+    --ha \
+    --addon-config=addon-config.yaml
+```
+
+#### 2. Pre-flight checks
+
+Before the test suite is run, the following actions shall be carried out in sequence:
+1. Check if `kubectl` binary is available
+2. Check if `linkerd` binary is available
+3. Check if connection to cluster can be established via `kubectl`
+4. Check if Linkerd control plane is installed
+5. Install the sample application
+
+Once the test suite is complete (success / failure ) all resources created for the test are destroyed if not immediately after each test.
```

Comment:
I prefer to keep the resources around if the tests failed so I can debug them.

Reply:
Hmm. But what if the tests are successful? I think it would be nice to automatically clean up the resources in that case. However, is it possible that sometimes users may just want to clean up anyway regardless of the failure of success of the tests? We could then have a `--clean` flag that indicates that the resources must always be cleaned.